python/examples/token_swap/token_swap.py

```diff
@@ -0,0 +1,370 @@
+
+    tx_proposer_send_token = ctx.tx[sf_amount]
+    tx_approver_send_token = ctx.tx[sf_amount_other] 
+    
+    if tx_approver_send_token.is_xrp() or tx_proposer_send_token.is_xrp():
+        return temBAD_AMOUNT
+
+    if not ctx.tx.is_field_present(sf_expiration):
+        return temBAD_EXPIRATION
+
+
+    return preflight2(ctx)
+
+def do_apply_propose(ctx, _mPriorBalance, _mSourceBalance):
+    
+    print("Token Swap Create test doApply...")
+
+    tx_approver_send_token = ctx.tx[sf_amount]
+    tx_proposer_send_token = ctx.tx[sf_amount_other]
+    tx_account = ctx.tx[sf_account]
+    tx_account_other = ctx.tx[sf_account_other]
+    tx_expiration = ctx.tx[sf_expiration]
+
+    account = ctx.view().peek(account_keylet(tx_account))
+    acc_seq = int(str(ctx.tx.get_seq_proxy().value()), 16)
+
+    # Token that the proposer is willing to receive
+    proposer_send_token_issue = tx_proposer_send_token.issuer()
+    # Token that the proposer is willing to provide
+    proposer_receive_token_issue = tx_approver_send_token.issuer()
+
+    close_time = ctx.view().info().parent_close_time
+
+    if expired(tx_expiration, close_time):
+        return tecEXPIRED
+
+    trustline_proposer_sending_token = ctx.view().peek(trustline_keylet(tx_account, proposer_send_token_issue))
+    trustline_receiving_sending_token = ctx.view().peek(trustline_keylet(tx_account, proposer_receive_token_issue))
+    trustline_approver_sending_token = ctx.view().peek(trustline_keylet(tx_account_other, proposer_receive_token_issue))
+    trustline_receiving_sending_token = ctx.view().peek(trustline_keylet(tx_account_other, proposer_send_token_issue))
+
+    # Check trustlines exist
+    if not trustline_proposer_sending_token or not trustline_receiving_sending_token or not trustline_approver_sending_token or not trustline_receiving_sending_token:
+        return tecNO_LINE
+    
+    token_swap_keylet = new_token_swap_keylet(tx_account, acc_seq)
+
+    slep = make_sle(token_swap_keylet)
+    slep[sf_account] = tx_account
+    slep[sf_amount] = tx_approver_send_token
+    slep[sf_token_swap_id] = acc_seq
+    slep[sf_amount_other] = tx_proposer_send_token
+    slep[sf_account_other] = tx_account_other
+    slep[sf_expiration] = tx_expiration
+
+    ctx.view().insert(slep)
+
+    page = ctx.view().dir_insert(tx_account, token_swap_keylet)
+    
+    if page is None:
+        return tecDIR_FULL
+    
+    slep[sf_owner_node] = page
+
+    adjust_owner_count(ctx.view(), account, 1, ctx.journal)
+
+    return tesSUCCESS
+
+def preflight_accept(ctx):
+
+    print("Token Swap Accept test preflight...")
+
+    amount = ctx.tx[sf_amount]
+    amountOther = ctx.tx[sf_amount_other]
+
+    if amount.is_xrp() or amountOther.is_xrp():
+        return temBAD_AMOUNT
+    
+    return preflight2(ctx)
+
+def do_apply_accept(ctx, _mPriorBalance, _mSourceBalance):
+
+    print("Token Swap Accept test doApply...")
+
+    tx_approver_send_token = ctx.tx[sf_amount]
```

Comment:
Is it possible to have shorter names for all of these variables? It's a bit difficult to read the code with the long variable names.